lib/model/model.go

```diff
@@ -1907,15 +1894,18 @@ func (m *model) DownloadProgress(device protocol.DeviceID, folder string, update
 }
 
 func (m *model) deviceStatRef(deviceID protocol.DeviceID) *stats.DeviceStatisticsReference {
-	m.fmut.Lock()
-	defer m.fmut.Unlock()
-
-	if sr, ok := m.deviceStatRefs[deviceID]; ok {
+	m.fmut.RLock()
```

Comment:
This is wrong as concurrent calls might both conclude there is no ref and create a new one, clobbering each other. The original is correct.

Reply:
Right. My beaf with this is that an informational call can block everything else due to taking that lock. In https://github.com/syncthing/syncthing/pull/5975/commits/df1ffe610049957195cc9ae877399b3bc9379288 I moved the creation of `deviceStatRefs` into `CommitConfiguration` such that we only require read-locks here.